src/main/java/org/nuxeo/wopi/lock/LockHelper.java

```diff
@@ -0,0 +1,169 @@
+            try (Session session = openLockDirectorySession()) {
+                DocumentModel entry = session.getEntry(fileId);
+                entry.setProperty(LOCK_DIRECTORY_SCHEMA_NAME, LOCK_DIRECTORY_LOCK, lock);
+                entry.setProperty(LOCK_DIRECTORY_SCHEMA_NAME, LOCK_DIRECTORY_TIMESTAMP, Instant.now().toEpochMilli());
+                session.updateEntry(entry);
+            }
+        });
+    }
+
+    /**
+     * Updates the WOPI lock stored for the given file id with a fresh timestamp.
+     */
+    public static void refreshLock(String fileId) {
+        Framework.doPrivileged(() -> {
+            try (Session session = openLockDirectorySession()) {
+                DocumentModel entry = session.getEntry(fileId);
+                entry.setProperty(LOCK_DIRECTORY_SCHEMA_NAME, LOCK_DIRECTORY_TIMESTAMP, Instant.now().toEpochMilli());
+                session.updateEntry(entry);
+            }
+        });
+    }
+
+    /**
+     * Removes the WOPI lock stored for the given file id.
+     */
+    public static void removeLock(String fileId) {
+        Framework.doPrivileged(() -> {
+            try (Session session = openLockDirectorySession()) {
+                session.deleteEntry(fileId);
+            }
+        });
+    }
+
+    /**
+     * Returns the list of expired stored WOPI locks according to the {@link Constants#LOCK_TTL} for each repository.
+     */
+    public static Map<String, List<DocumentModel>> getExpiredLocksByRepository() {
+        return Framework.doPrivileged(() -> {
+            try (Session session = openLockDirectorySession()) {
+                return Framework.getService(RepositoryManager.class)
+                                .getRepositoryNames()
+                                .stream()
+                                .map(repository -> getExpiredLocks(session, repository))
+                                .reduce(new HashMap<String, List<DocumentModel>>(), (a, b) -> {
+                                    a.putAll(b);
+                                    return a;
+                                });
+            }
+        });
+    }
+
+    protected static Map<String, List<DocumentModel>> getExpiredLocks(Session session, String repository) {
+        List<DocumentModel> expiredLocks = session.query(
+                Collections.singletonMap(LOCK_DIRECTORY_REPOSITORY, repository))
+                                                  .stream()
+                                                  .filter(LockHelper::filterExpiredLocks)
+                                                  .collect(Collectors.toList());
+        return Collections.singletonMap(repository, expiredLocks);
+    }
+
+    protected static boolean filterExpiredLocks(DocumentModel entry) {
+        return Instant.now()
+                      .minus(LOCK_TTL, ChronoUnit.MINUTES)
+                      .toEpochMilli() > (long) entry.getProperty(LOCK_DIRECTORY_SCHEMA_NAME, LOCK_DIRECTORY_TIMESTAMP);
```

Comment:
You should compute it only once, not on every filter call.
`System.currentTimeMills() - TimeUnit.MINUTES.toMillis(LOCK_TTL)` is simpler.
Also you could keep LOCK_TTL expressed in milliseconds as before to avoid minutes/milliseconds conversions here.